sdk/inc/azure/core/az_mqtt5_telemetry_consumer_codec.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition of #az_mqtt5_telemetry_consumer_codec.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_MQTT5_TELEMETRY_CONSUMER_CODEC_H
+#define _az_MQTT5_TELEMETRY_CONSUMER_CODEC_H
+
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief MQTT5 Telemetry Consumer Codec options.
+ *
+ */
+typedef struct
+{
+  /**
```

Comment:
Should this (and the corresponding producer one) have service_group_id as a field?